test/smoketest_artman.py

```diff
@@ -51,13 +52,27 @@ def run_smoke_test(root_dir, log):
     for artman_yaml_path in glob.glob('%s/google/**/artman_*.yaml' % root_dir,
                                       recursive=True):
         artman_config = _parse(artman_yaml_path)
+        # Reorder the artifacts to move the gapic_config generation to the end.
+        # Otherwise, the following GAPIC library generation is going to use the
+        # newly-generated GAPIC config instead of the one in github, leading to
+        # false negative and false positive.
+        artifacts = []
+        gapic_config_artifact = None
         for artifact in artman_config.artifacts:
+            if config_pb2.Artifact.GAPIC_CONFIG == artifact.type:
```

Comment:
nit: in case of multiple `GAPIC_CONFIG` artifacts:

`artifacts = sorted(artman_config.artifacts, key=lambda a: config_pb2.Artifact.GAPIC_CONFIG == a.type))`

?

Reply:
Super nice